src/client/client.ts

```diff
@@ -85,14 +120,17 @@ export default class TaskClient {
     account: string,
     database: string,
     collection: string,
-    key: string,
+    connectionInfo: {
```

Comment:
since this is a breaking change anyway, i wouldn't try to encapsulate the credential this way. there's no indication from the types that you need to provide exactly one of these options. i would do one of two things instead:

1. break `create` into `createFromKey` or `createFromCredential` with the appropriate type for each
2. just make this param `credential: string | TokenCredential` and then just check the type internally

(meta note: `ChainedTokenCredential` is overly specific. just take a `TokenCredential` and let people pass in whatever type of credential you want)